src/Identity/Extensions.Core/src/UserManager.cs

```diff
@@ -2276,7 +2277,60 @@ public virtual string GenerateNewAuthenticatorKey()
     /// </summary>
     /// <returns></returns>
     protected virtual string CreateTwoFactorRecoveryCode()
-        => Guid.NewGuid().ToString().Substring(0, 8);
+    {
+        var recoveryCode = new StringBuilder();
+        recoveryCode.Append(GetRandomRecoveryCodeChar());
+        recoveryCode.Append(GetRandomRecoveryCodeChar());
+        recoveryCode.Append(GetRandomRecoveryCodeChar());
+        recoveryCode.Append(GetRandomRecoveryCodeChar());
+        recoveryCode.Append(GetRandomRecoveryCodeChar());
+        recoveryCode.Append('-');
+        recoveryCode.Append(GetRandomRecoveryCodeChar());
+        recoveryCode.Append(GetRandomRecoveryCodeChar());
+        recoveryCode.Append(GetRandomRecoveryCodeChar());
+        recoveryCode.Append(GetRandomRecoveryCodeChar());
+        recoveryCode.Append(GetRandomRecoveryCodeChar());
+        return recoveryCode.ToString();
+    }
+
+    private static char GetRandomRecoveryCodeChar()
+    {
+        // We don't want to use any confusing characters like 0/O 1/I/L/l
+        // Taken from windows valid product key source
+        const string AllowedChars = "23456789BCDFGHJKMNPQRTVWXY";
```

Comment:
I understand some confusing chars were removed, in OC we also use them as lower-case which I find even more explicit. But I have the feeling upper case is more standard.